script/preprocessing.py

```diff
@@ -111,11 +115,42 @@ def musdb(file_path):
             else:
                 sound_class = None
 
-        else:
-            instrument_name = None
-            sound_class = None
-
-    return tempo, instrument_name, key, sound_class            
-
 
+    return tempo, instrument_name, key, sound_class
 
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+            prog="PreprocessingHelper",
+            description="This script creates metadata for BRID and/or MUSDB"
+            )
+
+    # arguments. required --> data home, dataset (if using preprocessing)
+    parser.add_argument("--data_home", required=True, help="pathway to where is data is stored")
+    parser.add_argument("--dataset", required=True, help="supported datasets: BRID (enter 'brid') and MUSDB (enter 'musdb')")
+    parser.add_argument("--track_files", help="txt file with track names")
+    # need to develop this still
+
+    args = parser.parse_args()
+    kwargs = vars(args)
+
+    if args.dataset == "brid":
+        for root, dirs, files in os.walk(args.data_home):
+            for file in files:
+                file_path = os.path.join(root, file)
+                args.tempo, args.instrument_name, args.key, args.sound_class = brid(file_path)
+
+                # extracting unique metadata for all brid .wav files
+                if file_path.endswith(".wav") or file_path.endswith(".mp3"):
+                    metadata.extraction(file_path, **kwargs)
+    elif args.dataset == "musdb":
+        for root, dirs, files in os.walk(args.data_home):
+            for file in files:
+                file_path = os.path.join(root, file)
+                args.tempo, args.instrument_name, args.key, args.sound_class = musdb(file_path)
+                print("instrument name ",args.instrument_name)
+
+                # extracting unique metadata for all musdb .wav files
+                if file_path.endswith(".wav") or file_path.endswith(".mp3"):
+                    metadata.extraction(file_path, **kwargs)
```

Comment:
we're doing exactly the same thing in the two blocks. let's use a dictionary to map `dataset` to function and remove the `if/elif/else` statements.